File: packages/types/src/generation.ts
/**
 * Generation types for PhotoGenius AI
 */

export type GenerationMode =
  | "realism"
  | "creative"
  | "romantic"
  | "cinematic"
  | "fashion"
  | "cool_edgy"
  | "artistic"
  | "max_surprise";
export type GenerationStatus = "pending" | "preview" | "complete" | "failed" | "blocked";

export interface GenerationRequest {
  prompt: string;
  mode: GenerationMode;
  identityId?: string;
  preset?: string;
  numOutputs?: number;
}

export interface GenerationResult {
  id: string;
  previewUrl?: string;
  imageUrls: string[];
  selectedUrl?: string;
  qualityReport?: QualityReport;
  status: GenerationStatus;
  blockedReason?: string;
}

export interface QualityReport {
  overallScore: number;        // 0-100
  faceMatchPercent?: number;   // 0-100, null if no identity
  aestheticScore: number;      // 0-10 (LAION scale)
  technicalQuality: number;    // 0-100
  promptAdherence: number;     // 0-100 (CLIP similarity)
}

export interface PreviewResponse {
  previewImage: string;
  message: string;
  estimatedTime: string;
  qualityLevel: string;
}

export interface FullQualityResponse {
  finalImage: string;
  message: string;
  qualityReport?: QualityReport;
}

File: apps/ai-service/app/models/generation.py
from pydantic import BaseModel  # type: ignore[reportMissingImports]
from typing import Literal


class GenerationRequest(BaseModel):
    prompt: str
    mode: Literal["realism", "creative", "romantic"] = "realism"
    identity_id: str | None = None
    preset: str | None = None
    num_outputs: int = 1


class PreviewResponse(BaseModel):
    preview_image: str
    message: str
    estimated_time: str
    quality_level: str


class FullQualityResponse(BaseModel):
    final_image: str
    message: str


class V1GenerateRequest(BaseModel):
    """Request body for POST /api/v1/generate."""
    identity_id: str
    prompt: str
    mode: Literal["realism", "creative", "romantic"] = "realism"
    user_id: str


class V1GenerateResponse(BaseModel):
    """Response for POST /api/v1/generate."""
    images: list[str]
    error: bool = False
    violations: list[str] | None = None
